peps/pep-0752.rst

```diff
@@ -0,0 +1,399 @@
+namespace MUST be prefixed by the root grant namespace followed by a hyphen.
+For example, ``google-cloud`` would be a valid child of the root ``google``.
+
+.. _grant-ownership:
+
+Grant Ownership
+---------------
+
+The owner of a grant may allow the grant to be shared with any number of other
+organizations. Organizations with which grants are shared cannot themselves
+share such grants. For example, if organization A was given a root grant and
+decides to share this with organization B:
+
+* Organization B cannot share the grant with other organizations.
+* Organization B cannot produce a child grant.
+
+The owner may also transfer this executive ownership to another organization.
+If the organization is a corporate organization, the target for transfer must
+also be.
+
+.. _uploads:
+
+Uploads
+-------
+
+If the following criteria are all true for a given upload:
+
+1. The package does not yet exist.
+2. The name matches a reserved namespace.
+3. The user is not authorized to use the namespace by the owner of the
+   namespace.
+
+Then the upload MUST fail with a 403 HTTP status code.
+
+.. _user-interface:
+
+User Interface
+--------------
+
+Every page for a particular release
+(`example <https://pypi.org/project/google-cloud-compute/1.19.2/>`__)
+that both matches an active namespace grant and is tied to an
+`owner <grant-ownership_>`_
+MUST receive a special indicator that signifies this tie. The UI also MUST
+indicate what the prefix is (NuGet does not do this). Repositories SHOULD have
+a dedicated page that enumerates every active namespace grant and which
+organization(s) own it.
+
+.. _public-namespaces:
+
+Public Namespaces
+-----------------
+
+The owner of a grant may choose to allow others the ability to release new
+packages with the associated namespace. Doing so MUST allow
+`uploads <uploads_>`_ for new packages matching the namespace from any user
+but such releases MUST NOT have the `visual indicator <user-interface_>`_.
+
+Root grants given to community projects SHALL always be public.
+
+.. _repository-metadata:
+
+Repository Metadata
+-------------------
+
+To allow installers and other tooling insight into this metadata for a given
+artifact upload of a namespaced package, the :pep:`JSON API <691>` MUST include
+the following keys:
+
+* ``namespace``: This is the associated `normalized <naming_>`_
+  namespace e.g. ``foo-bar``. If the namespace matches a child grant and the
+  user happens to be authorized for both the child and the root grant, this
+  MUST be the namespace associated with the child grant.
+* ``owner``: This is the organization with which the user is associated and
+  owner of the grant. If the namespace is `public <public-namespaces_>`_ and
+  the user is not part of the owner organization, this key MUST NOT be present.
+  This omission is useful for tools that wish to make a distinction between
+  official and community packages.
```

Comment:
Is it better for the key to be `null` rather than not present?

Reply:
I'm not sure about the terminology distinction between "official" and "community" packages.  Perhaps "organization owned" and "contributor"?

Reply:
Also, as above, I think it could be useful to both be able to link from a community package to the granting organization, and to know that the package is community owned.  Can we add some keys that make this explicit, rather than just omitting a key or mailing it `null`?

Reply:
Rather than `null` or nonexistent I changed it to `__public__`, good idea!